system/ee/EllisLab/ExpressionEngine/Service/Addon/Addon.php

```diff
@@ -47,13 +47,11 @@ public function __call($fn, $args)
 	 */
 	public function isInstalled()
 	{
-		$types = array('modules', 'fieldtypes', 'extensions');
-
 		ee()->load->model('addons_model');
 
-
 		if (is_null(self::$installed_modules))
 		{
+			/** @var \CI_DB_result $query */
 			$query = ee()->addons_model->get_installed_modules();
```

Comment:
@kevincupp I wonder if there is a better way to do this. Other systems have had issues with the global whatever() function, which may be the root cause of this b/c some of these methods do have the correct return type in the doc blocks, but the method chain may be breaking down?

Reply:
Yeah I'd prefer not having to put a comment everywhere something like this is used. What's the problem, exactly? Your IDE doesn't know the return type of `ee()`? Or it can't handle chaining methods like this?